.github/workflows/build_package.yml

```diff
@@ -131,29 +130,98 @@ jobs:
         if: "matrix.build-family != 'windows'"
         shell: bash
         run: |
-          cat << EOF > ./c/version_info.json
+          echo "meta package"
+          cat << EOF > ./c/version_info_rc.json
           {
             "package-suffix": "${{ github.event.inputs.package_suffix }}",
             "package-version": "${{ github.event.inputs.package_version }}",
             "iree-revision": "$(cd ./c && git rev-parse HEAD)"
           }
           EOF
-          cat ./c/version_info.json
+          cat ./c/version_info_rc.json
+
+          echo "iree-base-compiler"
+          cat << EOF > ./c/compiler/version_info_rc.json
+          {
+            "package-suffix": "${{ github.event.inputs.package_suffix }}",
+            "package-version": "${{ github.event.inputs.compiler_package_version }}",
+            "iree-revision": "$(cd ./c && git rev-parse HEAD)"
+          }
+          EOF
+          cat ./c/compiler/version_info_rc.json
+
+          echo "iree-base-runtime"
+          cat << EOF > ./c/runtime/version_info_rc.json
+          {
+            "package-suffix": "${{ github.event.inputs.package_suffix }}",
+            "package-version": "${{ github.event.inputs.runtime_package_version }}",
+            "iree-revision": "$(cd ./c && git rev-parse HEAD)"
+          }
+          EOF
+          cat ./c/runtime/version_info_rc.json
+
+          echo "iree-tools-tf"
+          cat << EOF > ./c/integrations/tensorflow/python_projects/iree_tf/version_info_rc.json
+          {
+            "package-suffix": "${{ github.event.inputs.package_suffix }}",
+            "package-version": "${{ github.event.inputs.tf_package_version }}",
+            "iree-revision": "$(cd ./c && git rev-parse HEAD)"
+          }
+          EOF
+          cat ./c/integrations/tensorflow/python_projects/iree_tf/version_info_rc.json
+
+          echo "iree-tools-tflite"
+          cat << EOF > ./c/integrations/tensorflow/python_projects/iree_tflite/version_info_rc.json
+          {
+            "package-suffix": "${{ github.event.inputs.package_suffix }}",
+            "package-version": "${{ github.event.inputs.tflite_package_version }}",
+            "iree-revision": "$(cd ./c && git rev-parse HEAD)"
+          }
+          EOF
+          cat ./c/integrations/tensorflow/python_projects/iree_tflite/version_info_rc.json
+
 
       - name: Write version info Windows (release)
         if: "matrix.build-family == 'windows'"
         shell: powershell
```

Comment:
The fork between powershell on Windows and bash on Linux/macOS was why I added a `setup_metadata` job in SHARK-Platform: https://github.com/nod-ai/SHARK-Platform/blob/46debb409130e8d9442701c4c06607ee7416b33a/.github/workflows/build_packages.yml#L16-L48

Probably worth doing something similar here, given how non-trivial the powershell is.